src/psygnal/_group.py

```diff
@@ -45,11 +45,30 @@ class EmissionInfo(NamedTuple):
     Attributes
     ----------
     signal : SignalInstance
+        The SignalInstance doing the emitting
     args: tuple
+        The args that were emitted
+    loc: str | int | None | tuple[int | str, ...]
+        If the emitter was a `SignalGroup` attribute on another object, this
+        will be the location of that emitter on the parent (e.g. name of that attribute
+        or index if the parent was an evented sequence).  Otherwise, it will be `None`.
+        If this is a flattened EmissionInfo, then this will be a tuple of locations.
     """
 
     signal: SignalInstance
     args: tuple[Any, ...]
+    loc: str | int | None | tuple[int | str, ...] = None
```

Comment:
Looking at the earlier example, which had `attr_name` here, I find that easier to think about than `loc`. This is especially true if you add dictionaries into the mix — so now we don't know from the type whether this is an index, a key, or an attribute name. What do you think about having three optionals:
- attr
- index
- key

Of course, now I move on to read about "flatten"... 😅 

You could make a lightweight "Loc" object that could contain each of the keys, then you could have a sequence of Loc objects? The nice thing about this option is that you could write an accessor that gets you the right object given the parent and the sequence of locs.

I'm also not super keen on the "loc" name. "from"? "emitted_by"? "child"?

Reply:
What about "origin"?

I think "loc" would be fine though...